server/src/main/java/io/druid/server/router/TieredBrokerHostSelector.java

```diff
@@ -312,11 +313,18 @@ Server pick()
         return null;
       }
 
-      if (roundRobinIndex >= currNodes.size()) {
-        roundRobinIndex %= currNodes.size();
+      int index = roundRobinIndex.get();
+
+      while (true) {
+        int nextIndex = index + 1;
+        if (nextIndex < 0) nextIndex %= currNodes.size();
```

Comment:
It won't help, the result of mod operation when the first argument is negative is zero or negative